### PR TITLE
Add support for Fulcio and Rekor, and --sign-by-sigstore=param-file

### PR DESCRIPTION
**Pull request by mtrmac**
`skopeo copy` and `skopeo sync` now support `--sign-by-sigstore=param-file`,
using the `containers-sigstore-signing-params.yaml(5)` file format.

That notably adds support for Fulcio and Rekor signing.

Depends on unmerged https://github.com/containers/image/pull/1787 ; see that PR for documentation of the YAML file format, as well as example files.

Interoperability with `cosign` tested manually, similarly to those examples. Integration tests to come later, tracked in #1704 .

(Yes, this is costly in binary size. On macOS, the binary goes from 27 MB to 33 MB. That can almost certainly be brought down by not using the official client packages, at the cost of replicating (the relevant subset of) their functionality.)

### CODE REVIEW

**Comment by mtrmac** [comment]
@vrothberg PTAL. @rhatdan FYI.

**Comment by rhatdan** [comment]
Awesome work @mtrmac

File: docs/skopeo-copy.1.md [hunk 1]
```diff
@@ -93,6 +93,11 @@ Do not copy signatures, if any, from _source-image_. Necessary when copying a si
 
 Add a “simple signing” signature using that key ID for an image name corresponding to _destination-image_
 
+**--sign-by-sigstore** _param-file_
+
+Add a sigstore signature based on further optinos specified in a containers sigstore signing parameter file _param-file_.
```
Comment (TomSweeneyRedHat) [inline docs/skopeo-copy.1.md:98]:
```suggestion
Add a sigstore signature based on further options specified in a container's sigstore signing parameter file _param-file_.
```

File: docs/skopeo-sync.1.md [hunk 2]
```diff
@@ -76,6 +76,11 @@ Print usage statement.
 
 Add a “simple signing” signature using that key ID for an image name corresponding to _destination-image_
 
+**--sign-by-sigstore** _param-file_
+
+Add a sigstore signature based on further optinos specified in a containers sigstore signing parameter file _param-file_.
```
Comment (TomSweeneyRedHat) [inline docs/skopeo-sync.1.md:81]:
```suggestion
Add a sigstore signature based on further options specified in a container's sigstore signing parameter file _param-file_.
```

**Comment by TomSweeneyRedHat** [comment]
Couple doc nits, otherwise LGTM.
Nicely done, I was hoping you'd keep this PR under 967 changed or new files!  ;^)

File: docs/skopeo-copy.1.md (hunk 1 above)
Comment (mtrmac) [reply docs/skopeo-copy.1.md]: Thanks, fixed the `options` typo.

WRT the other part, this is not “a parameter file owned by / related to a container”, it’s “a signing parameter file defined by the `GitHub.com/containers` organization (i.e. not a `cosign`-defined parameter file).

I fully appreciate that a “containers sigstore signing parameter file” is a horrible mouthful that desperately needs replacing; what would be a better name? Compare a tiny bit more discussion in https://github.com/containers/image/pull/1787 . Ultimately the name for this concept is decided by the man page in that other PR, and this one would follow the lead.

Comment (TomSweeneyRedHat) [reply docs/skopeo-copy.1.md]: Oh, that's entertaining.  Thanks for the explanation.  I'll try another whack at it.

File: docs/skopeo-copy.1.md [hunk 3]
```diff
@@ -93,6 +93,11 @@ Do not copy signatures, if any, from _source-image_. Necessary when copying a si
 
 Add a “simple signing” signature using that key ID for an image name corresponding to _destination-image_
 
+**--sign-by-sigstore** _param-file_
+
+Add a sigstore signature based on further options specified in a containers sigstore signing parameter file _param-file_.
```
Comment (TomSweeneyRedHat) [inline docs/skopeo-copy.1.md:98]: Perhaps?
```suggestion
Add a sigstore signature based on the options in the specified containers sigstore signing parameter file, _param-file_.
```
```

File: docs/skopeo-copy.1.md (hunk 1 above)
Comment (mtrmac) [reply docs/skopeo-copy.1.md]: “entertaining”. Seriously, I would love to replace that name.

File: docs/skopeo-copy.1.md (hunk 3 above)
Comment (mtrmac) [reply docs/skopeo-copy.1.md]: Thanks, updated.

**Review by vrothberg — COMMENTED** [review]
LGTM

**Comment by mtrmac** [comment]
Now based on a merged c/image feature, ready for review and possible merging.

**Comment by rhatdan** [comment]
LGTM

**Review by vrothberg — APPROVED** [review]
/lgtm

**Comment by rhatdan** [comment]
I probably did it by accident, all I wanted to do was prevent force push to main.

**Pull request merged**